impl/src/main/java/org/jboss/weld/event/ObserverNotifier.java

```diff
@@ -427,22 +427,24 @@ private NotificationMode initModeOption(Object value) {
      * @param notifyAction
      * @return a new supplier
      */
-    private <T, U extends T> Supplier<T> createSupplier(SecurityContext securityContext, U event, EventMetadata metadata, ObserverExceptionHandler exceptionHandler,
+    private <T, U extends T> Supplier<T> createSupplier(Consumer<Runnable> securityContextActionConsumer, U event, EventMetadata metadata, ObserverExceptionHandler exceptionHandler,
             boolean handleExceptions, Runnable notifyAction) {
         return () -> {
             final ThreadLocalStackReference<EventMetadata> stack = currentEventMetadata.pushIfNotNull(metadata);
             final RequestContext requestContext = requestContextHolder.get();
-            try {
-                securityContext.associate();
-                requestContext.activate();
-                notifyAction.run();
-            } finally {
-                stack.pop();
-                requestContext.invalidate();
-                requestContext.deactivate();
-                securityContext.dissociate();
-                securityContext.close();
-            }
+            securityContextActionConsumer.accept(() -> {
+                try {
+                    requestContext.activate();
+                    System.out.println(Thread.currentThread().getId());
```

Comment:
Uh my bad, didn't realize I created a PR already. Just wanted to store it somewhere :-D
Will remove of course.